Source/Core/DiscIO/VolumeWii.cpp

```diff
@@ -252,9 +253,8 @@ bool VolumeWii::HasWiiEncryption() const
 
 std::vector<Partition> VolumeWii::GetPartitions() const
 {
-  std::vector<Partition> partitions;
-  for (const auto& pair : m_partitions)
-    partitions.push_back(pair.first);
+  std::vector<Partition> partitions(m_partitions.size());
+  std::ranges::copy(m_partitions | std::views::keys, partitions.begin());
   return partitions;
```

Comment:
One way to partially address the readability concern and the hypothetical follow-up PR  would be to add an intermediary variable. AFAICT, it seems to generate similar code without optimization:
```suggestion
  std::vector<Partition> partitions(m_partitions.size());
  const auto partitions_view = std::views::keys(m_partitions);
  std::ranges::copy(partitions_view, partitions.begin());
  return partitions;
```